### ISSUE TITLE
Implement SNS message publish over http

### ISSUE THREAD

**Issue by mariusgrigaitis** [p4tin/goaws#75]
https://github.com/p4tin/goaws/blob/master/gosns/gosns.go#L276

Currently messages only published to SQS queues. Would be a lot more useful if there would be an ability to post messages to http endpoint.

**Comment by mariusgrigaitis** [comment]
This is some mock implementation of similar thing

https://github.com/yourkarma/fake_sns/blob/master/lib/fake_sns/deliver_message.rb#L81

**Comment by jasdel** [comment]
hi @mariusgrigaitis thanks for contacting us. If you're looking to specify a custom endpoint for the SQS or SNS service clients you can easily do this using the `aws.Config`'s `Endpoint`. This overrides the endpoint the SDK would be sending the request to. Does this help with the issue you're running into?  The messages will still be sent as if they were being sent to AWS. You can disable needing to use credentials by setting the `aws.Config.Credentials` field to `credentials.Anonymous`.

```go
    sess := session.Must(session.NewSession())
    sqsSvc := sqs.New(sess, &aws.Config{
        Endpoint: aws.String("custom_endpoint"),
        DisableSSL: aws.Bool(true), // Optionally disable SSL if the endpoint is http.
    })

    sqs.SendMessage(...)
```

**Comment by jasdel** [comment]
opps sorry comment on thread by mistake :)

**Comment by p4tin** [comment]
ok I will look at an http publish!!

**Comment by zikphil** [comment]
Any updates on this?

**Comment by AdrieanKhisbe** [comment]
Any news on this?

**Comment by p4tin** [comment]
This was implemented under PR #175
